libs/common/src/lib/combine-reducers.ts

```diff
@@ -0,0 +1,50 @@
+// Credits go to NgRx
+// Copied from with small modifications: https://github.com/ngrx/platform/blob/12.0.0/modules/store/src/utils.ts
+
+// The MIT License (MIT)
+//
+// Copyright (c) 2017 Brandon Roberts, Mike Ryan, Victor Savkin, Rob Wormald
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import { Action, Reducer, ReducerDictionary } from './models';
+
+export function combineReducers<T>(reducers: ReducerDictionary<T>): Reducer<T>;
+export function combineReducers(reducers: ReducerDictionary<any>): Reducer<any> {
+    const reducerKeys: string[] = Object.keys(reducers);
+    const reducerKeyLength: number = reducerKeys.length;
+
+    return (state: any = {}, action: Action) => {
+        const stateKeysLength: number = Object.keys(state).length;
+
+        let hasChanged: boolean = stateKeysLength !== reducerKeyLength;
+        const nextState: any = {};
+
+        for (let i = 0; i < reducerKeyLength; i++) {
+            const key = reducerKeys[i];
+            const reducer: any = reducers[key];
```

Comment:
Is there a reason why this is explicitly typed as `any` ? Seems unnecessary.

Reply:
Reducer<any> is indeed better than any.